rdkPlugins/Storage/source/BindLoopMountDetails.cpp

```diff
@@ -0,0 +1,459 @@
+        }
+    }
+
+    AI_LOG_FN_EXIT();
+    return success;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Decrement the reference count for data.img file.
+ */
+void BindLoopMountDetails::decrementReferenceCount()
+{
+    AI_LOG_FN_ENTRY();
+
+    std::unique_ptr<RefCountFile> refCountFile = getRefCountFile();
+
+    if (refCountFile)
+    {
+        RefCountFileLock lock(refCountFile);
+
+        if (refCountFile->Decrement() == 0) {
+            unmountLoopbackDevice();
+        }
+    }
+    
+    AI_LOG_FN_EXIT();
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Checks if image should be non persistent and if so remove it.
+ *
+ *  @return true on success, false on failure.
+ */
+bool BindLoopMountDetails::removeNonPersistentImage()
+{
+    AI_LOG_FN_ENTRY();
+
+    bool success = true;
+
+    // There could be situation where container was created but not started, as
+    // the cleanup is done after container start it would mean that we could
+    // left temp directories mounted. As postStop is done always we need to
+    // check if we have already cleaned up, and if not then clean temp.
+    if (access(mTempMountPointOutsideContainer.c_str(), F_OK) != -1)
+    {
+        success = cleanupTempDirectory();
+    }
+
+    auto refCountFile = getRefCountFile();
+
+    if (refCountFile)
+    {
+        RefCountFileLock lock(refCountFile);
+        // If reference count is zero, unmount the loopback device
+        if (refCountFile->Decrement() == 0)
+        {
+            unmountLoopbackDevice();
+        }
+
+        if (!mMount.persistent)
+        {
+            if (unlink(mMount.fsImagePath.c_str()))
+            {
+                AI_LOG_SYS_ERROR(errno, "failed to delete image file @ '%s'",
+                                mMount.fsImagePath.c_str());
+                success = false;
+            }
+            else
+            {
+                AI_LOG_DEBUG("Unlinked successfully @ '%s'", mMount.fsImagePath.c_str());
+            }
+        }
+    }
+    else
+    {
+        success = false;
+    }
+    
+    AI_LOG_FN_EXIT();
+    return success;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Open the reference count file for the data.img file.
+ *
+ *  @return unique_ptr to reference count file, nullptr on failure.
+ */
+std::unique_ptr<RefCountFile> BindLoopMountDetails::getRefCountFile()
+{
+    AI_LOG_FN_ENTRY();
+
+    struct stat file_stat;
+    int ret = stat(mMount.fsImagePath.c_str(), &file_stat);
+
+    if (ret < 0) {  
+        AI_LOG_ERROR("failed to get file stat for file '%s'",
+                     mMount.fsImagePath.c_str());
+        return nullptr;
+    }
+
+    // Create reference count file based on unique inode of data.img file
+    std::string refCountFilePath = std::string("/tmp/") + std::to_string(file_stat.st_ino);
```

Comment:
Rather than just in `/tmp`, this should be in the Dobby workspace directory (there use to be a `IDobbyEnv::workspaceMountPath()` call to get the workspace path).  Or it that doesn't exist anymore then suggest prefixing name with something like `/tmp/dobbyLoopDeviceRef_<inode_number>`

Reply:
I don't think I can call `IDobbyEnv::workspaceMountPath()` inside the Storage plugin. So used this format instead: /tmp/dobbyLoopDeviceRef_<inode_number>